airflow/models/taskinstance.py

```diff
@@ -550,7 +550,7 @@ def _xcom_pull(
         if first is None:  # No matching XCom at all.
             return default
         if map_indexes is not None or first.map_index < 0:
-            return XComModel.deserialize_value(first)
+            return first.value
```

Comment:
But this is causing us to special case it in https://github.com/apache/airflow/pull/47918/files#r2001428628